beet/test/compat/enums.ts

```diff
@@ -37,3 +41,25 @@ test('compat enums: milligrams', (t) => {
   }
   t.end()
 })
+
+test('compat enums: directions using int', (t) => {
+  const beet = fixedScalarEnum(Directions)
+  const buf = Buffer.alloc(beet.byteSize)
+  beet.write(buf, 0, Directions.Down)
+  // Clearer test to show expected values
+  checkFixedSerialize(
+    t,
+    beet,
+    Directions.Down,
+    [0x02],
+    'Directions.Down Serialize'
+  )
+  checkFixedDeserialize(
+    t,
+    beet,
+    Directions[2] as unknown as Directions,
```

Comment:
The fact that we have to case to `unknown` here makes me afraid that now this won't work with solita generated code out of the box?
Why is this suddenly necessary when it wasn't before and we didn't change anything about the enum type annotations nor the FixedBeet created from it?